amr-wind/utilities/sampling/SamplingUtils.cpp

```diff
@@ -0,0 +1,187 @@
+#include "SamplingUtils.H"
+#include "AMReX_ParmParse.H"
+
+namespace amr_wind::sampling::sampling_utils {
+
+vs::Vector reflect(vs::Vector line, vs::Vector vec)
+{
+
+    vs::Tensor ref(
+        1 - 2 * line.x() * line.x(), -2 * line.x() * line.y(),
+        -2 * line.x() * line.z(), -2 * line.y() * line.x(),
+        1 - 2 * line.y() * line.y(), -2 * line.y() * line.z(),
+        -2 * line.z() * line.x(), -2 * line.z() * line.y(),
+        1 - 2 * line.z() * line.z());
+
+    return vec & ref;
+}
+
+vs::Vector rotate_euler_vec(vs::Vector axis, double angle, vs::Vector vec)
+{
+
+    axis.normalize();
+    const auto RotMat = vs::quaternion(axis, angle);
+    return vec & RotMat;
+}
+
+vs::Vector rotate_euler_vector(vs::Vector& axis, double& angle, vs::Vector& vec)
+{
+    axis.normalize();
+    const auto RotMat = vs::quaternion(axis, angle);
+    return vec & RotMat;
+}
+
+vs::Vector rotation(const vs::Vector& angles, const vs::Vector& data)
+{
+    const vs::Tensor rotMatrix =
+        vs::xrot(angles.x()) & vs::yrot(angles.y()) & vs::zrot(angles.z());
+    return data & rotMatrix;
+}
+
+vs::Vector canon_rotator(const vs::Vector& angles, const vs::Vector& data)
+{
+    // Note: Angles in degrees as per "vs" class
+    const vs::Tensor rotMatrix =
+        vs::xrot(angles.x()) & vs::yrot(angles.y()) & vs::zrot(angles.z());
+    return data & rotMatrix;
+}
+
+vs::Tensor unit_projection_matrix(const vs::Vector& a)
+{
+    return {a[0] * a[0], a[0] * a[1], a[0] * a[2], a[0] * a[1], a[1] * a[1],
+            a[1] * a[2], a[0] * a[2], a[1] * a[2], a[2] * a[2]};
+}
+
+vs::Tensor rotation_matrix(vs::Vector dst, vs::Vector src)
+{
+    auto vmat = skew_cross(dst, src);
+    const auto ang = dst & src;
+
+    const double small = 1e-14 * vs::mag(dst);
+    if (std::abs(1 + ang) < small) {
+        return scale(vs::Tensor::I(), -1);
+    }
+    return vs::Tensor::I() + vmat + scale((vmat & vmat), 1. / (1 + ang));
+}
+
+vs::Tensor skew_cross(vs::Vector a, vs::Vector b)
+{
+    auto cross = b ^ a;
+    return {0,         -cross[2], cross[1], cross[2], 0,
+            -cross[0], -cross[1], cross[0], 0};
+}
+
+vs::Tensor scale(vs::Tensor v, double a)
+{
+    vs::Tensor vnew;
+    for (int j = 0; j < 9; ++j) {
+        vnew[j] = a * v[j];
+    }
+    return vnew;
+}
+
+void spherical_cap_quadrature(
```

Comment:
There is a lot of code added in SamplingUtils, and these standalone functions seem like good candidates for unit testing. Don't need tests for every single one, but ideally if there is a basic problem with a known solution for a few of the complicated ones, it would help solidify this part of the code and rule out errors here if there is debugging required in the future.